### PR TITLE
Add SFTP Service

### LINKED ISSUE #104: Integrate sftp-service
We're ready to soft launch the [SFTP service](https://github.com/PermanentOrg/sftp-service) as it exists today.

Unlike other typescript services, this will be deployed to its own machine / image rather than being packaged as part of the overall "backend" image.  Dev ssh access will also be done over a non-standard port (not port 22) since the user-facing service needs to be on port 22.

### PR DESCRIPTION
**Pull request by slifty**
This PR adds support for configuring and deploying the SFTP service to Permanent infrastructure

Resolves #104

### CODE REVIEW

File: provisioners/configure-sftp.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+echo "Install packages"
+apt -qq update
+apt -qq install -y \
+  curl \
+  wget \
+  nginx \
+	nginx-extras \
+	unzip \
+	jq
```
Comment (slifty) [inline provisioners/configure-sftp.sh:28]: This is not actually used any more

**Comment by slifty** [comment]
I'm not quite sure what Terraform is checking for and what needs to be done to resolve the failing checks; maybe tomorrow I can chat with someone who has more knowledge about that part of the pipeline and fix it up!

**Comment by slifty** [comment]
@cecilia-donnelly would it be appropriate to manually run the `publish dev` github action from this branch to make sure sftp is properly deployed?

File: .env.template [hunk 2]
```diff
@@ -63,3 +63,11 @@ export NOTIFICATION_FIREBASE_CREDENTIALS='{"format":"minified JSON";"see":"https
...
 export NEW_RELIC_APPLICATION_NAME=Backend-dev
 export NEW_RELIC_LICENSE_KEY=
+
+# FusionAuth credentials
+# See https://fusionauth.io/docs/v1/tech/apis/api-keys
+export FUSION_AUTH_HOST=${FUSION_AUTH_HOST}
+export FUSION_AUTH_KEY=${FUSION_AUTH_KEY}
+
+# The location of the permanent server (e.g. "dev.permanent.org")
+export SERVER_NAME=${SERVER_NAME}
```
Comment (liam-lloyd) [inline .env.template:73]: Is `SERVER_NAME` the best thing to call this? (`SERVER_DOMAIN` comes to mind as another candidate)

File: provisioners/deploy-sftp.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+---
+- name: Deploy code on machine and configure cron and services
```
Comment (liam-lloyd) [inline provisioners/deploy-sftp.yml:2]: Possible leftover from copy/paste here?

File: .env.template (hunk 2 above)
Comment (slifty) [reply .env.template]: Thank you!  Fixed.

**Comment by slifty** [comment]
I was able to deploy this via terraform, but I think port 22222 is blocked (which makes sense!) so it's not possible to SSH to the machines; I'm going to see how to update the utilized security group.

**Comment by slifty** [comment]
I edited the Developer / Production / Staging security groups on AWS to allow for port traffic on 22222 and I think we're set there.

File: .github/workflows/build-images.yml [hunk 4]
```diff
@@ -67,6 +76,9 @@ jobs:
           NOTIFICATION_FIREBASE_CREDENTIALS: ${{ secrets[matrix.environment.notification_firebase_credentials] }}
           SENTRY_DSN: ${{ secrets.SENTRY_TOKEN }}
           PERM_ENV: ${{ matrix.environment.perm_env }}
+          SERVER_DOMAIN: ${{ matrix.environment.server_domain }}
+          FUSION_AUTH_HOST: ${{ matrix.environment.fusion_auth_host }}
```
[the remark this answers is not in the document]
Comment (slifty) [reply .github/workflows/build-images.yml:80]: I take it back, it is not intuitive to me that a matrix environment value called `fusion_auth_host` would actually contain the secret key index holding the real value.  I'm going to try to just populate the environment matrix with the intended value directly and see if that works.

[the remark this answers is not in the document]
Comment (slifty) [reply .github/workflows/build-images.yml]: One more update: github didn't like that, so I've renamed it to be explicit that the matrix value holds a secret key.

File: README.md [hunk 5]
```diff
@@ -63,7 +66,7 @@ In both cases, the AMIs needs to be rebuilt and deployed. See [Workflow](#Workfl
 
 ## Debian Version
 
-We use the latest Debian base image published [here](https://wiki.debian.org/Cloud/AmazonEC2Image/Buster) for our builds.
+We use the latest Debian base image perscribed in [our debian verison policy](docs/POLICIES.md).
```
Comment (slifty) [inline README.md:69]: Typo: `perscribed` should be `prescribed`

**Comment by slifty** [comment]
All righty!  So this was a little odd to test since Terraform looks to `main` to determine how to deploy (and part of this PR is updating the terraform commands).  That said, I was able to temporarily test *dev* by updating the branch it pointed to be this one.

That worked!

So basically I:

1) Ran the build images action in github against this branch, which passes now
2) Ran the terraform run against this branch, which resulted in a deployed AMI that I could SSH to as a dev and connect to via STFP.

I think this is ready for re-review!

**Review by liam-lloyd — APPROVED** [review]

**Pull request merged**